conans/client/profile_loader.py

```diff
@@ -147,6 +147,9 @@ def _load_profile(text, profile_path, default_folder):
         # Replace the variables from parents in the current profile
         profile_parser.apply_vars()
 
+        # Expand env vars after variable substitution
+        profile_parser.profile_text = os.path.expandvars(profile_parser.profile_text)
```

Comment:
This sounds a bit risky, to be honest. It is a ``os.path`` operation, so at some point it might do some assumptions about characters, whitespace including carriage return, etc.